planner/cascades/transformation_rules.go

```diff
@@ -1286,3 +1288,38 @@ func (r *MergeAdjacentSelection) OnTransform(old *memo.ExprIter) (newExprs []*me
 	newSelExpr.SetChildren(childGroups...)
 	return []*memo.GroupExpr{newSelExpr}, true, false, nil
 }
+
+// MergeAdjacentLimit merge the adjacent limit.
+type MergeAdjacentLimit struct {
+	baseRule
+}
+
+// NewRuleMergeAdjacentLimit creates a new Transformation MergeAdjacentLimit.
+// The pattern of this rule is `Limit->Limit->X`.
+func NewRuleMergeAdjacentLimit() Transformation {
+	rule := &MergeAdjacentLimit{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandLimit,
+		memo.EngineAll,
+		memo.NewPattern(memo.OperandLimit, memo.EngineAll),
+	)
+	return rule
+}
+
+// OnTransform implements Transformation interface.
+// This rule tries to merge adjacent limit, with no simplification.
+func (r *MergeAdjacentLimit) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
+	limit := old.GetExpr().ExprNode.(*plannercore.LogicalLimit)
+	child := old.Children[0].GetExpr().ExprNode.(*plannercore.LogicalLimit)
+	childGroups := old.Children[0].GetExpr().Children
+
+	offset := child.Offset + limit.Offset
+	count := uint64(math.Min(float64(child.Count-limit.Offset), float64(limit.Count)))
```

Comment:
What if `child.Count < limit.Offset`? For example, `select a from (select a from t limit 3 offset 5) t1 limit 3 offset 5;`
Could we convert it to a `TableDual`?

Reply:
why not make a new rule to convert  `TableDual`, like  `OptimizeLimitZero` of spark sql.
may be a sql like `select xxx limit x ` and x <= 0.
and we can convert  `TableDual` in one rule

Reply:
Make sence. Could you please help implement the rule `TransformLimtiToTableDual` in another PR?

But I still concern about `min(child.Count - limit.Offset), limit.Count)`, if `child.Count < limit.Offset` it will return a negative `Count`.

Reply:
ok,I will take the initiative to convert it to `TableDual` in `MergeAdjacentLimit`.
and implement the rule `TransformLimtiToTableDual` to convert limit which can be converted to `TableDual`   in another PR.